package/libskylark/src/libskylark.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (C) 2016 Swift Navigation Inc.
+ * Contact: Jacob McNamee <jacob@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <curl/curl.h>
+#include <libsbp/observation.h>
+#include <libsbp/navigation.h>
+#include <string.h>
+#include <stdio.h>
+//#include <sbp_settings.h>
+#include <sbp_zmq.h>
+
+#include "libskylark.h"
+
+/**
+ *  Utilities
+ */
+
+// TODO (mookerji): Repeated here as is everwhere else!
+static int file_read_string(const char *filename, char *str, size_t str_size)
+{
+  FILE *fp = fopen(filename, "r");
+  if (fp == NULL) {
+    log_error("error opening %s\n", filename);
+    return -1;
+  }
+  bool success = (fgets(str, str_size, fp) != NULL);
+  fclose(fp);
+  if (!success) {
+    log_error("error reading %s\n", filename);
+    return -1;
+  }
+  return 0;
+}
+
+void log_client_config(const skylark_client_config_t *config) {
+  log_debug("client_config: endpoint_url=%s\n", config->endpoint_url);
+  log_debug("client_config: accept_type_header=%s\n", config->accept_type_header);
+  log_debug("client_config: content_type_header=%s\n", config->content_type_header);
+  log_debug("client_config: encoding=%s\n", config->encoding);
+  log_debug("client_config: device_header=%s\n", config->device_header);
+  log_debug("client_config: sbp_sender_id=%d\n", config->sbp_sender_id);
+  log_debug("client_config: enabled=%d\n", config->enabled);
+}
```

Comment:
I kept this largely for debugging purposes and thought it was useful. I think though that libcurl's verbose flag will also provide what's needed here, so I can probably remove it now.